draft-vkrasnov-h2-compression-dictionaries.md

```diff
@@ -138,58 +182,87 @@ The SET_COMPRESSION_CONTEXT frame (type=0xTBA).
 ~~~~
 
 The SET_COMPRESSION_CONTEXT frame can be sent by the client on any stream in
-the idle state (prior to the first HEADER frames on the stream).
-It indicates that the stream MAY be compressed by dictionaries with the given
-Context ID.
+the idle state. The frame indicates the compression context ID for the given
+stream. Frames with an assigned context SHALL NOT be compressed using
+dictionaries from a different context. Frames with an assigned context SHALL
+NOT be used as a dictionary for streams with from a different context.
 
 The SET_COMPRESSION_CONTEXT frame contains the following fields:
 
 Context:
-: an 8-bit Context ID that indicates the compression context for the stream.
-A stream can only be compressed by dictionaries with the same Context. If a
-stream SETs a dictionary, it will be assigned the Context indicated by this
-frame. If the frame is ommited, then the context value is assumed to be 0.
-The allowed Context values are 0 through 253.
-A special Context ID of 255 indicates the stream SHALL NOT be compressed at all
-and passed as is.
-A special Context ID of 254 indicates the stream SHALL NOT be compressed with
-any dictionary, and it MUST NOT be used as dictionary for other streams.
+: an 8-bit context ID that indicates the compression context for the stream.
+If the frame is ommited, then the context value is assumed to be 0.
+The allowed context values are 0 through 255.
+A special context ID of 255 indicates the stream SHALL NOT be cross-compressed.
```

Comment:
it's not clear if a `255` indication means that the stream can be compressed at the h2 layer, just not using a dictionary. I *think* that's the case, but might be worthwhile to make it clearer